### PR TITLE
sov: init at 0.63

### PR DESCRIPTION
**Pull request by twitchyliquid64**
###### Description of changes

Sov - Sway overview. Its super neat: https://www.reddit.com/r/swaywm/comments/ubsw9a/swayoverview_is_released/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by twitchyliquid64** [comment]
`nix-shell -p nixpkgs-review --run "nixpkgs-review pr 170459"`
```shell
[xxx@xxx:~/projects/nixpkgs]$ NIX_PATH="nixpkgs=./" nix-shell -p nixpkgs-review --run "nixpkgs-review pr 170459"
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/170459/head:refs/nixpkgs-review/1
From https://github.com/NixOS/nixpkgs
 + 8d43c4ad900...ab35e3e6139 refs/pull/170459/head -> refs/nixpkgs-review/1  (forced update)
$ git worktree add /home/xxx/.cache/nixpkgs-review/pr-170459-1/nixpkgs 1c99518c6f804c0d23ff82a20904768f18512909
Preparing worktree (detached HEAD 1c99518c6f8)
Updating files: 100% (30304/30304), done.
HEAD is now at 1c99518c6f8 Merge pull request #170444 from mdlayher/mdl-corerad-120
$ nix-env --option system x86_64-linux -f /home/xxx/.cache/nixpkgs-review/pr-170459-1/nixpkgs -qaP --xml --out-path --show-trace
$ git merge --no-commit --no-ff ab35e3e6139426146291319a2c4128b90833faba
Auto-merging pkgs/top-level/all-packages.nix
Automatic merge went well; stopped before committing as requested
$ nix-env --option system x86_64-linux -f /home/xxx/.cache/nixpkgs-review/pr-170459-1/nixpkgs -qaP --xml --out-path --show-trace --meta
1 package added:
sov (init at 2022-04-26)

warning: ignoring the user-specified setting 'system', because it is a restricted setting and you are not a trusted user
$ nix --experimental-features nix-command build --no-link --keep-going --option build-use-sandbox relaxed -f /home/xxx/.cache/nixpkgs-review/pr-170459-1/build.nix
warning: ignoring the user-specified setting 'sandbox', because it is a restricted setting and you are not a trusted user
building '/nix/store/i7md3aqvl6wy49jc17vwjga7akiqwybs-sov-unstable-2022-04-26.drv'...
building '/nix/store/bdzkcq046ylyaczq7yfcq8vfyz1ysz47-review-shell.drv'...

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/170459

1 package built:
sov

$ nix-shell /home/xxx/.cache/nixpkgs-review/pr-170459-1/shell.nix
```

**Comment by twitchyliquid64** [comment]
Hmmm not sure whats up with `aarch64-darwin` - this is Linux so that should be skipped?

File: pkgs/tools/wayland/sov/default.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "sov";
+  version = "unstable-2022-04-26";
+
+  src = fetchFromGitHub {
+    owner = "milgra";
+    repo = "sov";
+    rev = "5bed5299396eb5bb334fda83b8b84301ffa8ae51";
```
Comment (WolfangAukang) [inline pkgs/tools/wayland/sov/default.nix:22]: Any specific reason to not use [v0.63](https://github.com/milgra/sov/releases/tag/0.63)?

File: pkgs/tools/wayland/sov/default.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    sha256 = "01ynjx12dn7940nzw1x3c0i33p53an04x3ykniwki9cqd2ficg76";
+  };
+
+  # Need <linux/limits.h> in addition to <limits.h>
+  patchPhase = ''
+    find . -type f -iname "*.c" -exec sed -i.bak '/limits.h/a #include <linux/limits.h>' "{}" +;
+  '';
+
+  outPath = ["bin" "share" ];
```
Comment (WolfangAukang) [inline pkgs/tools/wayland/sov/default.nix:31]:
```suggestion
  outPath = [ "bin" "share" ];
```

File: pkgs/tools/wayland/sov/default.nix [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  nativeBuildInputs = [ meson ninja cmake pkg-config wayland-scanner ];
+  buildInputs = [
```
Comment (WolfangAukang) [inline pkgs/tools/wayland/sov/default.nix:34]:
```suggestion
  nativeBuildInputs = [
    meson
    ninja
    cmake
    pkg-config
    wayland-scanner
  ];
  
  buildInputs = [
```

File: pkgs/tools/wayland/sov/default.nix [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+  meta = with lib; {
+    homepage = "https://github.com/milgra/sov";
+    description = "An overlay that shows schemas for all workspaces to make navigation in sway easier.";
+    license = licenses.mit;
+    maintainers = with maintainers; [ twitchyliquid64 ];
+    platforms = platforms.linux;
+  };
+}
+
```
Comment (WolfangAukang) [inline pkgs/tools/wayland/sov/default.nix:53]: Please remove this extra line

**Review by WolfangAukang — COMMENTED** [review]
Some format recommendations

File: pkgs/tools/wayland/sov/default.nix (hunk 1 above)
Comment (twitchyliquid64) [reply pkgs/tools/wayland/sov/default.nix]: Nope, that was released after I made this PR. Updated!

**Comment by WolfangAukang** [comment]
Result of `nixpkgs-review pr 170459` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sov</li>
  </ul>
</details>

**Comment by TravisDavis-ops** [comment]
Heads up just merged version 0.71 of this

**Comment by WolfangAukang** [comment]
#174492

**Pull request closed without merging**